src/System.Private.CoreLib/shared/System/AggregateException.cs

```diff
@@ -438,16 +438,12 @@ public override string ToString()
             StringBuilder text = new StringBuilder();
             text.Append(base.ToString());
 
-            for (int i = 0; i < m_innerExceptions.Count; i++)
+            foreach (Exception ex in m_innerExceptions)
             {
-                if (m_innerExceptions[i] == InnerException)
+                if (object.ReferenceEquals(ex, InnerException))
                     continue; // Already logged in base.ToString()
 
-                text.Append(Environment.NewLineConst + InnerExceptionPrefix);
-                text.AppendFormat(CultureInfo.InvariantCulture, SR.AggregateException_InnerException, i);
-                text.Append(m_innerExceptions[i].ToString());
-                text.Append("<---");
-                text.AppendLine();
+                text.Append(InnerExceptionToString(ex));
```

Comment:
This helper ends up allocating it's own string builder, formatting to it, getting its string, and then appending that string to this builder. Why not just pass this builder in? If the concern is reusing it in places that don't use a builder, you can make the helper create the builder only if one want passed in, and return a its ToString only if one was.

Reply:
Sure.